src/raftstore/store/worker/region.rs

```diff
@@ -253,6 +252,7 @@ impl<T: MsgSender> Runner<T> {
                     box_try!(self.db.write(wb));
                     break;
                 }
+                box_try!(util::check_key_in_region(keys::origin_key(&key), &region));
```

Comment:
If the snapfile is not correct, any key is possible to be decoded from the file.